app/Http/Controllers/Api/PostController.php

```diff
@@ -75,4 +75,32 @@ public function getCommentsOfPost($id)
         $comments = Comment::getParentComments($id);
         return metaResponse($comments);
     }
+
+    /**
+     * Get list post of user
+     *
+     * @param Request $request request
+     * @param int     $id      id of user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getListPostOfUser(Request $request, $id)
+    {
+        $fields = [
+            'books.name as book_name',
+            'books.image',
+            'books.avg_rating',
+            'posts.updated_at',
+        ];
+        $userId = User::findorFail($id)->id;
+        $postTypes = config('define.type_post');
+        $checkType = in_array($request->type, $postTypes);
```

Comment:
cái này em check ở Request được mà nhỉ, dùng `in_array` hoặc `in` là ok, nhìn code sẽ dễ chịu hơn đó. https://laravel.com/docs/5.5/validation#rule-in-array